Don't encrypt `site-locale`

johnswanson opened the pull request:
Don't encrypt site locale. It's not a secret and doesn't need to be encrypted even if an encryption key is set.

johnswanson committed:
Don't encrypt `site-locale`

noahmoss commented:
Looks like there's a test failing which explicitly assumes that it can be encrypted...

johnswanson committed:
Fix test for `site-locale`

The previous test required encryption. Theoretically the referenced
error just depends on the fact that something is logged during the
process of getting the value, so I reworked it to depend on that
directly rather than triggering it through an encryption key rotation.
